sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -28,3 +29,45 @@ def get_token(self, *scopes, **kwargs):
     from collections import namedtuple
 
     AccessToken = namedtuple("AccessToken", ["token", "expires_on"])
+
+__all__ = ["AzureKeyCredential"]
+
+
+class AzureKeyCredential(object):
+    """Credential type used for authenticating to an Azure service.
+    It provides the ability to update the key without creating a new client.
+
+    :param str api_key: The API key to your Azure account.
+    :raises: TypeError
+    """
+
+    def __init__(self, api_key):
+        # type: (str) -> None
+        if not isinstance(api_key, six.string_types):
+            raise TypeError("api_key must be a string.")
+        self._api_key = api_key  # type: str
+
+    @property
+    def api_key(self):
+        # type () -> str
+        """The value of the configured API key.
+
+        :rtype: str
+        """
+        return self._api_key
+
+    def update_key(self, key):
```

Comment:
Consider changing to `update` for consistency with other languages

Reply:
Changed to `update`, thanks